core/utils/math_utils.hpp

```diff
@@ -76,31 +76,43 @@ inline bool approx_equals(double_t lhs, double_t rhs) noexcept {
   return std::fabs(rhs - lhs) < std::numeric_limits<double_t>::epsilon();
 }
 
+constexpr uint64_t ceil64(double_t v) noexcept {
```

Comment:
First of I don't understand why we need alias for double and float
Also don't understand why you think it's better than `std::ceil`
Or its because constexpr (since C++23 for std)? If yes maybe make them consteval?